### PR TITLE
libretro: Sync updates from libretro

### PR DESCRIPTION
**Pull request by RobLoach**
This brings in the latest changes from the libretro core. Includes updates to the build process in the Makefile.

### CODE REVIEW

File: libretro/jni/Application.mk [hunk 1]
```diff
@@ -1 +1,2 @@
+APP_STL := c++_static
```
Comment (LIJI32) [inline libretro/jni/Application.mk:1]: There's no need for this, SameBoy does not use C++.

File: libretro/Makefile [hunk 2]
```diff
@@ -119,6 +119,20 @@ else ifeq ($(platform), classic_armv7_a7)
...
+# Raspberry Pi 4 in 64 mode
+else ifneq (,$(findstring rpi4_64,$(platform)))
+   EXT ?= so
+   TARGET := $(TARGET_NAME)_libretro.$(EXT)
+   fpic := -fPIC
+   SHARED := -shared -Wl,--version-script=$(CORE_DIR)/libretro/link.T -Wl,--no-undefined
+   CFLAGS += -O2 -march=armv8-a+crc+simd -mtune=cortex-a72
+   CXXFLAGS += $(CFLAGS)
+   CPPFLAGS += $(CFLAGS)
+   ASFLAGS += $(CFLAGS)
```
Comment (LIJI32) [inline libretro/Makefile:133]: No need for AS/CXX/CPP flags, they're not used.

File: libretro/jni/Application.mk (hunk 1 above)
Comment (RobLoach) [reply libretro/jni/Application.mk:1]:
``` suggestion
```

File: libretro/Makefile (hunk 2 above)
Comment (RobLoach) [reply libretro/Makefile]: Thanks!

**Pull request merged**